### PR TITLE
Adds TRAIT_TRAPPED for support for oranges's new PR.

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request

Adds a prerequisite for https://github.com/tgstation/tgstation/pull/49500 @optimumtact 

## Why It's Good For The Game

It;s a compile prerequisite.

## Changelog
:cl:
fix: Fixes the compile. 
/:cl:

### CODE REVIEW

**Review by spookydonut — DISMISSED** [review]

**Comment by Tumesuo** [comment]
God yes speedmerge

**Review by Tumesuo — APPROVED** [review]

**Comment by Tharcoonvagh** [comment]
This is even worse

**Comment by trollbreeder** [comment]
What does this do?

**Review by spookydonut — DISMISSED** [review]

**Comment by Tharcoonvagh** [comment]
305e64

**Comment by trollbreeder** [comment]
Seems to be handcuff lite. Ah well. Speedmerge.

**Review by Tharcoonvagh — CHANGES_REQUESTED** [review]
make it gib you when you try to "trap" someone

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -236,6 +236,9 @@
 		return FALSE
 
 /mob/living/carbon/resist_buckle()
+	if (HAS_TRAIT(owner,TRAIT_TRAPPED))
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/carbon.dm]:
```suggestion
	if (HAS_TRAIT(src,TRAIT_TRAPPED))

```

File: code/modules/mob/living/carbon/carbon.dm [hunk 2]
```diff
@@ -236,6 +236,9 @@
 		return FALSE
 
 /mob/living/carbon/resist_buckle()
+	if (HAS_TRAIT(owner,TRAIT_TRAPPED))
+		to_chat(owner,"<span class='warning'>You're trapped and can't get out!</span>")
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/carbon.dm]:
```suggestion
		to_chat(src,"<span class='warning'>You're trapped and can't get out!</span>")

```

**Review by spookydonut — CHANGES_REQUESTED** [review]

**Pull request merged**